src/operator/contrib/transformer-inl.h

```diff
@@ -34,6 +34,19 @@
 namespace mxnet {
 namespace op {
 
+struct InterleavedMatMulParam : public dmlc::Parameter<InterleavedMatMulParam> {
+  int heads;
+  bool bwd_ignore_zero_init;
+  DMLC_DECLARE_PARAMETER(InterleavedMatMulParam) {
+    DMLC_DECLARE_FIELD(heads)
+    .describe("Set number of heads");
+    DMLC_DECLARE_FIELD(bwd_ignore_zero_init)
+    .describe("Make backward pass ignore AddTo and not init to 0. "
+              " /!\\ Only enable with MXNET_ENABLE_EXEC_ADDTO fonctionality")
```

Comment:
Is that constraint enforced and throws and error if the flag is not set?

Reply:
The problem is that MXNET_ENABLE_EXEC_ADDTO is used only during binding. So the user could potentially bind one symbol with MXNET_ENABLE_EXEC_ADDTO=0 then swap it to MXNET_ENABLE_EXEC_ADDTO=1 (for instance to bind another symbol). If I check for the flag both network will see it at 1, while only the 2nd network will actually use the functionality.

Reply:
Do you envision that to happen in regular use cases? In case you don't want to raise an exception, we could still print a warning in case the flag is not set

Reply:
It can happen, but probably not frequently. I can add a warning if you are fine with it.

Reply:
I found a way to raise an exception while cover my edge case. By checking with a custom ParamParser, as it's called during bindind. Hope the solution suit you :)

Reply:
Very neat solution, good job and thank you!